### PR TITLE
Hotfix/additional form options error case

### PR DESCRIPTION
**Pull request by carnage**
Reopened #7131 against develop

### CODE REVIEW

File: library/Zend/Form/Annotation/ElementAnnotationsListener.php [hunk 1]
```diff
@@ -163,6 +163,11 @@ public function handleComposedObjectAnnotation($e)
                 $specification['type'] = 'Zend\Form\Fieldset';
             }
 
+            if (isset($elementSpec['spec']['options'])) {
+                $specification['options'] = isset($specification['options']) ? $specification['options'] : array();
+                $specification['options'] = array_merge($elementSpec['spec']['options'], $specification['options']);
```
Comment (malukenho) [inline library/Zend/Form/Annotation/ElementAnnotationsListener.php]: Can't be
```php
if (isset($elementSpec['spec']['options']) && isset($specification['options'])) {
    $specification['options'] = array_merge($elementSpec['spec']['options'], $specification['options']);
}
```

??

Comment (weierophinney) [inline library/Zend/Form/Annotation/ElementAnnotationsListener.php]: @malukenho Good point; I'll take care of that on merge.

Comment (weierophinney) [inline library/Zend/Form/Annotation/ElementAnnotationsListener.php]: Actually, @malukenho, it's correct as specified; we're trying to set the `$specification['options']` and/or update it; if the conditional is changed as you specify, `$specification['options']` will be empty later.

Comment (malukenho) [inline library/Zend/Form/Annotation/ElementAnnotationsListener.php]: Thanks for explanation @weierophinney :)

**Pull request merged**